graph/src/log/elastic.rs

```diff
@@ -265,14 +267,16 @@ impl ElasticDrain {
                 batch_url.set_path("_bulk");
 
                 // Send batch of logs to Elasticsearch
-                let client = Client::new();
-
                 let header = match config.general.username {
-                    Some(username) => client
+                    Some(username) => config
+                        .general
+                        .client
```

Comment:
Shouldn't we use a `ClientBuilder` and bake the headers into the client instead of setting them with every request? I actually don't think it matters all that much. I had a brief look at the `reqwest` sources, and couldn't see how this change would avoid DNS requests.

Reply:
I think things are easier to follow if the headers are set where the request is made. On avoiding a DNS request, I'd assume that a new connection would generate a DNS request while reusing an existing one wouldn't. But not an expert here.